addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/handler/AccountHandler.java

```diff
@@ -0,0 +1,635 @@
+        BluetoothState state = null;
+
+        JsonBluetoothStates states = null;
+        if (temp.getIsLoggedIn()) {
+            states = temp.getBluetoothConnectionStates();
+        }
+
+        if (states != null) {
+            state = states.findStateByDevice(device);
+        }
+        echoHandler.updateState(device, state);
+    }
+
+    private void intializeChildDevice(@NonNull Connection connection, @NonNull EchoHandler child) {
+        Device deviceJson = this.findDeviceJson(child);
+        if (deviceJson != null) {
+            child.intialize(connection, deviceJson);
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+
+        cleanup();
+        super.handleRemoval();
+    }
+
+    @Override
+    public void childHandlerDisposed(@NonNull ThingHandler childHandler, @NonNull Thing childThing) {
+
+        // echo handler?
+        if (childHandler instanceof EchoHandler) {
+            synchronized (echoHandlers) {
+                echoHandlers.remove(childHandler);
+            }
+
+            AmazonEchoDiscovery instance = AmazonEchoDiscovery.instance;
+            if (instance != null) {
+                instance.removeExistingEchoHandler(childThing.getUID());
+            }
+        }
+
+        // flash briefing profile handler?
+        if (childHandler instanceof FlashBriefingProfileHandler) {
+            synchronized (flashBriefingProfileHandlers) {
+                flashBriefingProfileHandlers.remove(childHandler);
+            }
+        }
+
+        // smart home handler?
+        if (childHandler instanceof SmartHomeBaseHandler) {
+            synchronized (smartHomeHandlers) {
+                smartHomeHandlers.remove(childHandler);
+            }
+
+            AmazonEchoDiscovery instance = AmazonEchoDiscovery.instance;
+            if (instance != null) {
+                instance.removeExistingSmartHomeHandler(childThing.getUID());
+            }
+        }
+        super.childHandlerDisposed(childHandler, childThing);
+    }
+
+    @Override
+    public void dispose() {
+        LoginServlet loginServlet = this.loginServlet;
+        if (loginServlet != null) {
+            loginServlet.dispose();
+        }
+        this.loginServlet = null;
+        AmazonEchoDiscovery.removeDiscoveryHandler(this);
+        cleanup();
+        super.dispose();
+    }
+
+    private void cleanup() {
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+        if (refreshLogin != null) {
+            refreshLogin.cancel(true);
+            refreshLogin = null;
+        }
+        if (connection != null) {
+            connection.logout();
+            connection = null;
+        }
+    }
+
+    private void start() {
+        logger.debug("amazon account bridge starting handler ...");
+
+        config = getConfigAs(AccountConfiguration.class);
+        if (config.amazonSite == null || config.amazonSite.isEmpty()) {
```

Comment:
We as openHAB already have a depending to common.lang 2.6 it will make your code easier to read if your use there `StringUtils.isEmpty()`

Reply:
Sorry, I don't know this function. It's my first Java project (I'am normally a.NET developer). If found other issues like this, please let me know.